cmd/podman/machine/ssh.go

```diff
@@ -116,6 +116,11 @@ func ssh(cmd *cobra.Command, args []string) error {
 	}
 
 	username := sshOpts.Username
+
+	if mc.HostUser.Rootful {
+		username = "root"
+	}
+
```

Comment:
I suppose this below could be an `else if` now, because you always set username in this condition there is no need to check if username == "" for that branch again0.

Reply:
or combine all three into a switch where line 118 is the default @Luap99 ... i'm fine either way or as is ...

Reply:
let me know if the latest push is what you had in mind